### PR TITLE
libkml: bump to unreleased version for test suite fix

### PR DESCRIPTION
**Pull request by alexfmpe**
ZHF: #144627

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libkml/default.nix [hunk 1]
```diff
@@ -11,13 +11,13 @@
 
 stdenv.mkDerivation rec {
   pname = "libkml";
-  version = "1.3.0";
+  version = "unstable-2017-01-15"; # Last release is 1.3.0
```
Comment (alexfmpe) [inline pkgs/development/libraries/libkml/default.nix:14]: I suspect only https://github.com/libkml/libkml/commit/9163d288dfae64139563190dbf09f82ec28e8559 is actually needed, but applying that patch directly fails for some thunks. The other option would be hand-crafting a .patch file

Comment (SuperSandro2000) [reply pkgs/development/libraries/libkml/default.nix:14]:
```suggestion
  version = "unstable-2017-01-15";
```

File: pkgs/development/libraries/libkml/default.nix [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
 
   src = fetchFromGitHub {
     owner = "libkml";
     repo = pname;
-    rev = version;
-    sha256 = "0gl4cqfps9mzx6hzf3dc10hy5y8smpyf1s31sqm7w343hgsllv0z";
+    rev = "916a801ed3143ab82c07ec108bad271aa441da16";
+    sha256 = "sha256:0kgpmmc6dxnpqj90yplg3018d9cyykp6xd2cws49iv91w5k2c11v";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libkml/default.nix:20]:
```suggestion
    sha256 = "0kgpmmc6dxnpqj90yplg3018d9cyykp6xd2cws49iv91w5k2c11v";
```

**Comment by alexfmpe** [comment]
> I take it it's not feasible to just apply a patch to the tests rather than shipping an unofficial version?

https://github.com/NixOS/nixpkgs/pull/145071#discussion_r744733262

**Comment by risicle** [comment]
Sorry, those damned resolved comments.

Fails on macos 10.15 though:

```
[  5%] Linking CXX shared library ../../../lib/libkmlbase.dylib
Undefined symbols for architecture x86_64:
  "_crc32", referenced from:
      _libkml_unzReadCurrentFile in unzip.c.o
  "_get_crc_table", referenced from:
      _libkml_unzOpenCurrentFile3 in unzip.c.o
  "_inflate", referenced from:
      _libkml_unzReadCurrentFile in unzip.c.o
  "_inflateEnd", referenced from:
      _libkml_unzClose in unzip.c.o
      _libkml_unzCloseCurrentFile in unzip.c.o
      _libkml_unzOpenCurrentFile3 in unzip.c.o
      _libkml_unzDetach in unzip.c.o
  "_inflateInit2_", referenced from:
      _libkml_unzOpenCurrentFile3 in unzip.c.o
ld: symbol(s) not found for architecture x86_64
```

**Comment by thiagokokada** [comment]
So the situation:
- On our Tier 1 platform (`x86_64-linux`), this package fails since 2021-01-03
- If we update it, it builds on `x86_64-linux` but starts to fail on `x86_64-darwin` (that nowadays pass)
- Upstream seems abandoned
- This library is not dependency of anything on nixpkgs it seems

Maybe we could just remove it? Seems easier.

**Comment by thiagokokada** [comment]
> So the situation:
> 
>     * On our Tier 1 platform (`x86_64-linux`), this package fails since 2021-01-03 (so it is broken for at least one stable release)
> 
>     * If we update it, it builds on `x86_64-linux` but starts to fail on `x86_64-darwin` (that pass currently)
> 
>     * Upstream seems abandoned
> 
>     * This library is not dependency of anything on nixpkgs it seems
> 
> 
> Maybe we could just remove it? Seems easier.

Opened a PR removing this package: https://github.com/NixOS/nixpkgs/pull/145209

**Comment by veprbl** [comment]
The following fixes the darwin build
```
diff --git a/pkgs/development/libraries/libkml/default.nix b/pkgs/development/libraries/libkml/default.nix
index 8d04356769e..2ef1b13dc72 100644
--- a/pkgs/development/libraries/libkml/default.nix
+++ b/pkgs/development/libraries/libkml/default.nix
@@ -40,6 +40,8 @@ stdenv.mkDerivation rec {
     minizip
   ];
 
+  LDFLAGS = lib.optional stdenv.isDarwin "-lz";
+
   preCheck = ''
     export LD_LIBRARY_PATH=$PWD/lib
   '';
```
I suspect it would be needed to be set unconditionally, for it to not fail at runtime on Linux. Unfortunately we don't have a package to test that against.

**Comment by thiagokokada** [comment]
> I suspect it would be needed to be set unconditionally, for it to not fail at runtime on Linux. Unfortunately we don't have a package to test that against.

This is why I think it is better to simply remove it. We can't even make sure that this library is working properly because there is no other package that depends on it in nixpkgs.

**Comment by veprbl** [comment]
At least it passes its own test suite. Regarding deletion, we'd want to hear from the maintainer.

**Comment by thiagokokada** [comment]
> At least it passes its own test suite. Regarding deletion, we'd want to hear from the maintainer.

Yeah sure, let's wait from @costrouc until it answers what it thinks.

However we should always assume that the maintainer is no active anymore, and eventually take a decision even if it doesn't answer.

**Comment by thiagokokada** [comment]
Closed by https://github.com/NixOS/nixpkgs/pull/145209.

**Pull request closed without merging**